data-loader/src/index.ts

```diff
@@ -2,27 +2,148 @@ import { parse } from 'yaml';
+  operands: z.lazy(() => prerequisiteSchema.array()).optional(),
+});
+
+const prerequisitesSchema = z.array(prerequisiteSchema);
+
+function readSubjects() {
   const fileBuffer = readFileSync(SUBJECTS_FILE_PATH, 'utf8');
   const objects = Object.values(camelize(parse(fileBuffer) as Obj));
 
-  const subjects = subjectsSchema.parse(objects);
+  return subjectsSchema.parse(objects);
+}
+
+function readGroups() {
+  const fileBuffer = readFileSync(GROUPS_FILE_PATH, 'utf8');
+  const objects = Object.values(camelize(parse(fileBuffer) as Obj));
+
+  return groupsSchema.parse(objects);
+}
+
+function readPrerequisites() {
+  const fileBuffer = readFileSync(PREREQUISITES_FILE_PATH, 'utf8');
+  const objects = Object.values(camelize(parse(fileBuffer) as Obj));
+
+  return prerequisitesSchema.parse(objects);
+}
+
+// Goes over current prerequisite and creates the structure in the database. In is a recursive function.
+function savePrerequisites(prerequisite: Prerequisite, prevId?: string) {
+  // Create Previatura node related to previous or subject with unique id
+  // if logical, always create a new Previatura
+  // if subject, only relate to subject and return
+  // if credits, only relate to group or specify credits
+  
+  // For each operand, call savePrerequisite with prevId id of subject just created
+}
+
+async function main() {
+  const subjects = readSubjects();
+  const groups = readGroups();
+
+  const driver = getDriver().session();
+
+  console.log('Inserting ', subjects.length, ' subjects');
+
+  for (const [index, subject] of subjects.entries()) { 
+    const { name, code, credits, hasExam, subjectGroup } = subject;
+
+    await driver.run(
+      'CREATE (n: Materia { name: $name, code: $code, credits: $credits, hasExam: $hasExam, subjectGroup: $subjectGroup })',
+      {
+        name,
+        code,
+        credits: neo4j.int(credits),
+        hasExam: neo4j.int(+hasExam),
+        subjectGroup,
+      }
+    );
+
+    console.log('Inserted ', { name, code }, '---', index + 1, ' / ', subjects.length);
+  }
+
+  console.log('Inserting ', subjects.length, ' groups');
+  for (const [index, group] of groups.entries()) { 
+    const { name, code, minCredits } = group;
+
+    await driver.run(
+      'CREATE (n: Grupo { name: $name, code: $code, minCredits: $minCredits })',
+      {
+        name,
+        code,
+        minCredits: neo4j.int(minCredits),
+      }
+    );
+
+    console.log('Inserted ', { name, code }, '---', index + 1, ' / ', groups.length);
+  }
+
+  console.log('Relating subjects to groups');
+  for (const subject of subjects) {
+    const { code, subjectGroup } = subject;
+
+    if (!code || !subjectGroup) {
+      continue;
+    }
+
+    await driver.run(
+      `MATCH
+        (subject:Materia {code: $code}),
+        (group:Grupo {code: $subjectGroup})
+      CREATE (subject)-[:PERTENECE]->(group)`,
+      { code, subjectGroup },
+    );
+
+    console.log('Related ', { code, subjectGroup });
+  }
```

Comment:
I think you already said something about this, but just to leave it noted here. I think we need to create the groups before and associate the subjects when we create them. Also, I just saw that the subjects file has one group per subject, is that true on bedelias? I'll check that tomorrow but I thought that a subject could have more than one group